core/testsuite/src/main/basic/com/blazebit/persistence/testsuite/entity/JsonDocument.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2014 - 2020 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.blazebit.persistence.testsuite.entity;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+/**
+ * @author Moritz Becker
+ * @since 1.5.0
+ */
+@Entity
+@Table(name = "json_document")
+public class JsonDocument {
+    private Long id;
+    @Column(nullable = false, columnDefinition = "text")
```

Comment:
I believe this annotation is ignored(it should lead to an error as not all DBMS support the text datatype) because you already annotated the getter with `@Id` which causes the use of PROPERTY access, leading to ignoring this.

Also, this class is the same as the one in src/main/hibernate. Why have both? Apart from that, you are missing a default constructor.